crates/viewer/re_viewer_context/src/test_context.rs

```diff
@@ -72,11 +81,122 @@ impl Default for TestContext {
             reflection,
             command_sender,
             command_receiver,
+
+            // Created lazily since each egui_kittest harness needs a new one.
+            egui_render_state: Mutex::new(None),
         }
     }
 }
 
+fn create_egui_renderstate() -> Option<egui_wgpu::RenderState> {
+    re_tracing::profile_function!();
+
+    let shared_wgpu_setup = (*SHARED_WGPU_RENDERER_SETUP).as_ref()?;
+
+    let config = egui_wgpu::WgpuConfiguration {
+        wgpu_setup: egui_wgpu::WgpuSetup::Existing {
+            instance: shared_wgpu_setup.instance.clone(),
+            adapter: shared_wgpu_setup.adapter.clone(),
+            device: shared_wgpu_setup.device.clone(),
+            queue: shared_wgpu_setup.queue.clone(),
+        },
+
+        // None of these matter for tests as we're not going to draw to a surfaces.
+        present_mode: wgpu::PresentMode::Immediate,
+        desired_maximum_frame_latency: None,
+        on_surface_error: Arc::new(|_| {
+            unreachable!("tests aren't expected to draw to surfaces");
+        }),
+    };
+
+    let compatible_surface = None;
+    let depth_format = None;
+    let msaa_samples = 1;
```

Comment:
Don't we have `4` in the real app?

Reply:
re_renderer's individual views (managed each by a `ViewBuilder`) have, but egui's final target doesn't - re_Renderer resolves and copies into egui in `ViewBuilder::composite`
leaving a comment about this since this is quite the mental gymnastic!